src/DDTrace/Integrations/PDO/PDOSandboxedIntegration.php

```diff
@@ -0,0 +1,245 @@
+            PDOSandboxedIntegration::detectError($this, $span);
+        });
+
+        // public bool PDO::commit ( void )
+        dd_trace_method('PDO', 'commit', function (SpanData $span) {
+            if (dd_trace_tracer_is_limited()) {
+                return false;
+            }
+            $span->name = $span->resource = 'PDO.commit';
+            $span->service = 'PDO';
+            $span->type = Type::SQL;
+            PDOSandboxedIntegration::setConnectionTags($this, $span);
+        });
+
+        // public PDOStatement PDO::prepare ( string $statement [, array $driver_options = array() ] )
+        dd_trace_method('PDO', 'prepare', function (SpanData $span, array $args, $retval) {
+            if (dd_trace_tracer_is_limited()) {
+                return false;
+            }
+            $span->name = 'PDO.prepare';
+            $span->service = 'PDO';
+            $span->type = Type::SQL;
+            $span->resource = (string) $args[0];
+            PDOSandboxedIntegration::setConnectionTags($this, $span);
+            PDOSandboxedIntegration::storeStatementFromConnection($this, $retval);
+        });
+
+        // public bool PDOStatement::execute ([ array $input_parameters ] )
+        dd_trace_method('PDOStatement', 'execute', function (SpanData $span, array $args, $retval) use ($integration) {
+            if (dd_trace_tracer_is_limited()) {
+                return false;
+            }
+            $span->name = 'PDOStatement.execute';
+            $span->service = 'PDO';
+            $span->type = Type::SQL;
+            $span->resource = (string) $this->queryString;
+            if ($retval === true) {
+                $span->meta = [
+                    'db.rowcount' => (string) $this->rowCount(),
+                ];
+            }
+            PDOSandboxedIntegration::setStatementTags($this, $span);
+            $integration->addTraceAnalyticsIfEnabled($span);
+            PDOSandboxedIntegration::detectError($this, $span);
+        });
+
+        return SandboxedIntegration::LOADED;
+    }
+
+    /**
+     * @param \PDO|\PDOStatement $pdoOrStatement
+     * @param SpanData $span
+     */
+    public static function detectError($pdoOrStatement, SpanData $span)
+    {
+        $errorCode = $pdoOrStatement->errorCode();
+        // Error codes follows the ANSI SQL-92 convention of 5 total chars:
+        //   - 2 chars for class value
+        //   - 3 chars for subclass value
+        // Non error class values are: '00', '01', 'IM'
+        // @see: http://php.net/manual/en/pdo.errorcode.php
+        if (strlen($errorCode) !== 5) {
+            return;
+        }
+
+        $class = strtoupper(substr($errorCode, 0, 2));
+        if (in_array($class, ['00', '01', 'IM'], true)) {
+            // Not an error
+            return;
+        }
+        $errorInfo = $pdoOrStatement->errorInfo();
+        $span->meta[Tag::ERROR_MSG] = 'SQL error: ' . $errorCode . '. Driver error: ' . $errorInfo[1];
```

Comment:
What is the resulting interaction between automatic exception handling and setting this meta element?

Reply:
Auto-exception handling takes precedence. If no exception is thrown, a span can be marked as an error span if `error.msg` is set from userland. [Related comment](https://github.com/DataDog/dd-trace-php/pull/538/files#r321329487). :)